### ISSUE TITLE
Automatically consolidate autosave files

### ISSUE THREAD

**Issue by scanlime** [scanlime/robot-odyssey-rewired#9]
It's nice to have up-to-date save files almost always, but the resulting clutter in the UI and the database is a problem! We could solve this purely by filtering in the UI, but it would be nice to just automatically prune autosaves to a reasonable degree.

I'd like to avoid schemes which retain saves for a limited time unless we can very clearly and consentually communicate this with the player. Instead, maybe we can make an instantaneous decision about a save based on its neighbors in the database, so we can throw out redundant saves but keep autosaves that predate important game events, like changes in game item state or changing rooms.

**Comment by flintinatux** [comment]
I wouldn't mind being able to fully disable the autosave completely if desired, maybe just with a flag:
```
localStorage.AUTOSAVE_ENABLED = false
```
The built-in save mechanism in the game works fine for me, and wouldn't clutter the UI or the database.  I could stub out a PR for that if you'd like.

Could also add a configurable autosave interval to tone it down a bit if desired, and default the value to 500ms as before:
```
localStorage.AUTOSAVE_INTERVAL = 60000
```

**Comment by scanlime** [comment]
I think the nicest thing to do would be to consolidate "duplicate" autosaves, i.e. older saves that are in the same world perhaps. A configurable interval would be nice too. I'll try to make time to work on this, but a PR would be appreciated as well. Thanks!

**Comment by ZizzyZizzy** [comment]
Agreed. I came here to see if it was a bug that autosave happens every few seconds, making it near impossible to find the manual saves.

In addition to the above configuration options suggested, it would be fantastic if the manual saves were placed above the autosaves, making them easy to find in the sea of files.
